mixnet/connection.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import asyncio
+
+NetworkPacketQueue = asyncio.Queue[bytes]
+SimplexConnection = NetworkPacketQueue
+
+
+class DuplexConnection:
+    """
+    A duplex connection in which data can be transmitted and received simultaneously in both directions.
+    This is to mimic duplex communication in a real network (such as TCP or QUIC).
+    """
+
+    inbound: SimplexConnection
+    outbound: MixSimplexConnection
+
+    def __init__(self, inbound: SimplexConnection, outbound: MixSimplexConnection):
+        self.inbound = inbound
+        self.outbound = outbound
+
+    async def recv(self) -> bytes:
+        return await self.inbound.get()
+
+    async def send(self, packet: bytes):
+        await self.outbound.send(packet)
+
+
+class MixSimplexConnection:
+    """
+    Wraps a SimplexConnection to add a transmission rate and noise to the connection.
+    """
+
+    queue: NetworkPacketQueue
+    conn: SimplexConnection
+    transmission_rate_per_sec: int
+    noise_msg: bytes
```

Comment:
Likewise, why is this declared here?

If it is for typing, you can add it it to the `self` declarations below I think 🤔

Reply:
thank you e1abc44